torch/package/_glob_group.py

```diff
@@ -0,0 +1,44 @@
+from typing import Union, Iterable
+import re
+
+GlobPattern = Union[str, Iterable[str]]
+
+
+class _GlobGroup:
+    def __init__(self, include: "GlobPattern", exclude: "GlobPattern"):
+        self._dbg = f"_GlobGroup(include={include}, exclude={exclude})"
+        self.include = _GlobGroup._glob_list(include)
+        self.exclude = _GlobGroup._glob_list(exclude)
+
+    def __str__(self):
+        return self._dbg
+
+    def matches(self, candidate: str) -> bool:
+        if candidate[:5] != ".data":
```

Comment:
The only change I made to the _GlobGroup was preventing the `.data` from having the `.` prepended as it didn't allow the `.data` folder to be matched to

Reply:
This glob infrastructure is for package names, and assumes that things are separated by periods. But files are separated by `/` and might contain periods. I feel like this should just take the separate token as an argument to the GlobalGroup, and then this special case would not need to exist.

Reply:
Cool, I modified the GlobGroup to accept whatever separators. I can change the GlobGroup's `_glob_to_re` and `_glob_list` to not be staticmethods if that makes the code cleaner (I wasn't sure if they were staticmethods for a reason)